motionplan/constraint.go

```diff
@@ -301,9 +301,9 @@ func createAllCollisionConstraints(
 		return nil, err
 	}
 	return map[string]StateConstraint{
-		defaultObstacleConstraintName:       obstacleConstraint,
-		defaultSelfCollisionConstraintName:  selfCollisionConstraint,
-		defaultRobotCollisionConstraintName: robotConstraint,
+		defaultObstacleConstraintDesc:       obstacleConstraint,
+		defaultSelfCollisionConstraintDesc:  selfCollisionConstraint,
+		defaultRobotCollisionConstraintDesc: robotConstraint,
```

Comment:
potentially dumb question: does caching longer strings affect performance? @biotinker

Reply:
Broadly yes it can; when storing in a map, a longer string takes longer to run through whatever hash function. Plus larger sizes in bytes mean larger memory copies that take longer, if copies of a map are made.

At the scale of this change, and with the number of iterations of things that we run, we are likely several orders of magnitude away from noticing a difference.

> Theoretically, maps with keys from finite domains (such as ints) are trivially O(1) in space and time, and maps with keys with infinite domains (such as strings) require hashing and the specifics of equality testing to be included in the cost, which makes inserts and lookups best case O(N log N) on average (since the keys have to be at least O(log N) in size on average to construct a hash table with N entries.

From https://stackoverflow.com/questions/29677670/what-is-the-big-o-performance-of-maps-in-golang

Reply:
Yeah we aren't using this map a lot so I think this is fine, but something to be aware of I think

Reply:
An example of somewhere this *is* relevant is our RRT maps. Currently, it is a map of pointers to pointers, which is fast. However, this means that identical underlying node configurations are not only possible but common.

De-duplicating nodes in the RRT tree would involve actually parsing their contents rather than simply using an atomic memory address, and would have significant performance implications on path planning.

Additionally, this is why the `CachedTransform` function on model frames is not used; the performance penalty is sufficient that it is not consistently faster than simply recalculating forward kinematics.